service/controller/clusterapi/v21/resource/updatemachinedeployments/create.go

```diff
@@ -0,0 +1,97 @@
+package updatemachinedeployments
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/errors/tenant"
+	"github.com/giantswarm/microerror"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/kubernetes/pkg/apis/core"
+	"sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+
+	"github.com/giantswarm/cluster-operator/pkg/label"
+	"github.com/giantswarm/cluster-operator/service/controller/clusterapi/v21/key"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	cr, err := key.ToCluster(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	var machineDeployments []v1alpha1.MachineDeployment
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", "finding machine deployments for tenant cluster")
+
+		l := metav1.AddLabelToSelector(
+			&metav1.LabelSelector{},
+			label.Cluster,
+			key.ClusterID(&cr),
+		)
+		o := metav1.ListOptions{
+			LabelSelector: labels.Set(l.MatchLabels).String(),
+		}
+
+		list, err := r.cmaClient.ClusterV1alpha1().MachineDeployments(cr.Namespace).List(o)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		machineDeployments = list.Items
+
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("found %d machine deployments for tenant cluster", len(machineDeployments)))
+	}
+
+	var node core.Node
+	{
+		o := metav1.ListOptions{}
+		list, err := k8sClient.CoreV1().Nodes().List(o)
+		if tenant.IsAPINotAvailable(err) {
+			// fall through
+		} else if err != nil {
+			return nil, microerror.Mask(err)
+		}
+		if len(list.Items) == 0 {
+			// TODO return error handling
+		}
+
+		node = list.Items[0]
+	}
+
```

Comment:
```suggestion
	// versionLabel contains computed provider operator label for version bundle version update.
```
Abstract variable name puzzled me initially, but after realizing what happens here, I like it, but would appreciate comment. :slightly_smiling_face:

Reply:
Sorry this does not even compile right now. When pairing with @anvddriesch we noticed that there is a bit more to the whole label magic. She made a pretty nice suggestion which I am still implementing. Will get back here soon.

Reply:
Sorted